sidekick_core/lib/src/commands/format_command.dart

```diff
@@ -0,0 +1,158 @@
+  /// - Use `**/circle/**` to exclude `packages/circle` as well as
+  ///   `third_party/circle/packageA` and `third_party/circle/packageB`.
+  final List<String> excludeGlob;
+
+  FormatCommand({
+    this.excludeGlob = const [],
+  }) {
+    argParser.addOption(
+      'package',
+      abbr: 'p',
+    );
+    argParser.addOption(
+      'line-length',
+      abbr: 'l',
+    );
+    argParser.addFlag(
+      'verify',
+      help:
+          'Only verifies that all code is formatted, does not actually format it',
+    );
+  }
+
+  @override
+  Future<void> run() async {
+    final String? packageName = argResults?['package'] as String?;
+    final int? lineLength =
+        int.tryParse(argResults?['line-length'] as String? ?? '');
+    final bool verify = argResults?['verify'] as bool? ?? false;
+
+    final root = SidekickContext.projectRoot;
+    final List<DartPackage> allPackages = findAllPackages(root);
+    final globExcludes =
+        excludeGlob.map<Glob>((rule) => Glob("${root.path}/$rule"));
+    if (packageName != null) {
+      final package =
+          allPackages.where((it) => it.name == packageName).firstOrNull;
+      if (package == null) {
+        throw "Package with name $packageName not found in repository "
+            "${SidekickContext.repository?.path}";
+      }
+      final int lineLength = getLineLength(package);
+      final allDartFiles =
+          package.root.listSync(recursive: true).filterAllFiles(
+                globExcludes,
+              );
+      _format({lineLength: allDartFiles}, verify: verify);
+      return;
+    }
+
+    // Getting all Dart files exluding files which are starting with a .
+    final allFiles = SidekickContext.projectRoot
+        .listSync(recursive: true)
+        .filterAllFiles(
+          globExcludes,
+        )
+        .toList();
+
+    // Getting all directories excluding directories which are starting with a . and sort them by length
+    final sortedPackages =
+        allPackages.sortedByDescending((element) => element.root.path.length);
+
+    final lineLengthsAndFiles = <int, List<File>>{};
+
+    for (final package in sortedPackages) {
+      final resolvedLineLength = lineLength ?? getLineLength(package);
+      final filesInPackage = allFiles
+          .where((file) => file.path.contains(package.root.path))
+          .toList();
+      allFiles.removeWhere((file) => filesInPackage.contains(file));
+      if (filesInPackage.isNotEmpty) {
+        (lineLengthsAndFiles[resolvedLineLength] ??= []).addAll(filesInPackage);
+      }
+    }
+    _format(lineLengthsAndFiles, verify: verify);
+  }
+}
+
+extension on Iterable<FileSystemEntity> {
+  Iterable<File> filterAllFiles(Iterable<Glob> globExcludes) {
+    return whereType<File>()
+        .filter((file) => file.extension == '.dart')
+        .filter(
+          (file) => file.uri.pathSegments.none(
+            (element) => element.startsWith('.'),
+          ),
```

Comment:
This might fail when `projectRoot` itself is inside a hidden folder. Let's ignore this for now

Reply:
I did not mean to remove this check 😅

Reply:
Comment it out instead